src/app.js

```diff
@@ -363,16 +354,35 @@ class Messenger extends EventEmitter {
     // The 'payload' param is a developer-defined field which is set in a postback
     // button for Structured Messages.
     const payload = event.postback.payload;
-
     debug("onPostback for user:%d with payload '%s'", senderId, payload);
-
-    this.emit('text', new Response(this, {event, senderId, session, source: 'postback', text: payload}));
     this.emit('postback', new Response(this, {event, senderId, session, payload}));
+
+    if (this.emitOptionalEvents(event, senderId, session, payload)) {
+      return;
+    }
+    this.emit('text', new Response(this, {event, senderId, session, source: 'postback', text: payload, normalizedText: payload.toLowerCase().trim()}));
   }
 
   // HELPERS
   //////////
 
+  emitOptionalEvents(event, senderId, session, text) {
```

Comment:
I'd like to see a test case that shows the conditions that `emitOptionalEvents` will return true or false.